obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    else:
+        warnings.warn(mag_type + ' is not convertible')
+        return str('')
+    return str(mag)
+
+
+def _nortoevmag(mag_type):
+    """
+    Switch from nordic type magnitude notation to obspy event magnitudes.
+
+    >>> _nortoevmag('b')
+    'mB'
+    >>> _nortoevmag('bob')
+    ''
+    """
+    mag_type = str(mag_type)
+    if mag_type == 'L':
+        mag = 'ML'
+    elif mag_type == 'b':
+        mag = 'mB'
+    elif mag_type == 's':
+        mag = 'Ms'
+    elif mag_type == 'S':
+        mag = 'MS'
+    elif mag_type == 'W':
+        mag = 'MW'
+    elif mag_type == 'G':
+        mag = 'MbLg'
+    elif mag_type == 'C':
+        mag = 'Mc'
+    else:
+        warnings.warn(mag_type + ' is not convertible')
+        return str('')
+    return str(mag)
+
+
+def readheader(sfile):
+    """
+    Read header information from a seisan nordic format S-file.
+    Returns an obspy.core.event.Catalog type: note this changed for version \
+    0.1.0 from the inbuilt class types.
+
+    :type sfile: str
+    :param sfile: Path to the s-file
+
+    :returns: :class: obspy.core.event.Event
+    """
+    with open(sfile, 'r') as f:
+        header = _readheader(f=f)
+    return header
+
+
+def _readheader(f):
+    """Internal header reader.
+    :type f: file
+    :param f: File open in read-mode.
+
+    :returns: obspy.core.event.Event"""
+    f.seek(0)
+    # Base populate to allow for empty parts of file
+    new_event = Event()
+    topline = f.readline()
+    if not len(topline.rstrip()) == 80:
+        f.close()
+        raise IOError('s-file has a corrupt header, not 80 char long')
+    f.seek(0)
+    for line in f:
+        if line[79] in [' ', '1']:
+            topline = line
+            break
+        if line[79] == '7':
+            f.close()
+            raise IOError('No header found, corrupt s-file?')
+    try:
+        sfile_seconds = int(topline[16:18])
+        if sfile_seconds == 60:
+            sfile_seconds = 0
+            add_seconds = 60
+        else:
+            add_seconds = 0
+        new_event.origins.append(Origin())
+        new_event.origins[0].time = UTCDateTime(int(topline[1:5]),
+                                                int(topline[6:8]),
+                                                int(topline[8:10]),
+                                                int(topline[11:13]),
+                                                int(topline[13:15]),
+                                                sfile_seconds,
+                                                int(topline[19:20]) *
+                                                100000)\
+            + add_seconds
+    except:
+        f.close()
+        IOError("Couldn't read a date from sfile")
+    # new_event.loc_mod_ind=topline[20]
+    new_event.event_descriptions.append(EventDescription())
+    new_event.event_descriptions[0].text = topline[21:23]
+    # new_event.ev_id=topline[22]
+    if not _float_conv(topline[23:30]) == 999:
```

Comment:
`_float_conv` is not even useful in this case:
```python
try:
    new_event.origins[0].latitude = float(topline[23:30])
    new_event.origins[0].longitude = float(topline[31:38])
    new_event.origins[0].depth = float(topline[39:43]) * 1000
except ValueError:
    new_event.origins[0].latitude = None
    new_event.origins[0].longitude = None
    new_event.origins[0].depth = None
```
is perfectly clear as is, though as noted below, maybe don't even add the origin.